### DISCUSSION TITLE
How to use `go mod tidy` with `go_deps` when working with `.proto` files?

### DISCUSSION THREAD

**Discussion by Megakuul** [bazelbuild/bazel-gazelle#1827] [Q&A] [answered]
Hello, I have a question about how to use the `go_deps` gazelle extension in a project where I use proto files.

So I have a bzlmod project where I use `go` with grpc proto files. As I read [here](https://github.com/bazelbuild/rules_go/blob/master/docs/go/core/bzlmod.md#external-dependencies) it is recommended to manage external dependencies with the `go.mod` file and use the gazelle `go_deps` extension. I did this and everything works like expected, however if I try to update the dependencies with `bazel run @rules_go//go -- mod tidy -v` it does not find the imported proto file:
```
go: github.com/megakuul/cthulhu/shared/dbadapter/go imports
        github.com/megakuul/cthulhu/shared/dbadapter/rpc/etcdv3: no matching versions for query "latest"
```

Now, I'm confused how this problem can be solved because I would rather not precompile the proto files into my source tree (which would then make it work). I assume I could also add a `replace` directive in the `go.mod` file that points the proto imports to the `bazel-out` files, but this seems like a terrible idea to me.

Does anyone see what I'm doing wrong here or have an idea how to work with that correctly?

**Comment by jgautier-dd** [comment, marked as answer]
I found a discussion in rules_go that suggests to use:

```
bazel run @io_bazel_rules_go//go -- mod tidy -e
```

To ignore errors and continue trying to update the go.mod file.

Source: https://github.com/bazel-contrib/rules_go/issues/4104#issuecomment-2351835581

**Reply by Megakuul** [reply]
Thanks, even if that doesn't seem to be a very clean solution, it works. 

In the meantime, I have changed my project to include the protobuf output in the repository; I think this is generally the more appropriate approach for many projects, as it allows external programs outside the monorepo to depend on the precompiled api definitions.
